node_modules/oae-principals/tests/test-emails.js

```diff
@@ -0,0 +1,603 @@
+
+                // Sanity check the user has no `email` property yet
+                RestAPI.User.getUser(camAdminRestContext, user.id, function(err, user) {
+                    assert.ok(!err);
+                    assert.ok(!user.email);
+                    assert.strictEqual(user.emailVerified, false);
+                    assert.strictEqual(user.emailChange, email.toLowerCase());
+
+                    // Assert that there's no mapping yet for the email address
+                    PrincipalsTestUtil.assertUserEmailMapping(email, [], function() {
+
+                        // Verify the email address
+                        PrincipalsTestUtil.assertVerifyEmailSucceeds(anonymousRestContext, email, token, function() {
+
+                            // Assert the mapping has been created
+                            PrincipalsTestUtil.assertUserEmailMapping(email, [user.id], function() {
+                                return callback();
+                            });
+                        });
+                    });
+                });
+            });
+        });
+
+        /**
+         * Test that verifies that local users accounts created by a tenant administrator do not need to verify their email address
+         */
+        it('verify local user accounts created by a tenant administrator do not need to verify their email address', function(callback) {
+            // We can't use TestsUtil.generateTestUsers as that would do the verification process for us
+            var email = TestsUtil.generateTestEmailAddress();
+            var params = {
+                'username': TestsUtil.generateTestUserId(),
+                'password': 'password',
+                'displayName': TestsUtil.generateRandomText(1),
+                'email': email
+            };
+            PrincipalsTestUtil.assertCreateUserSucceeds(camAdminRestContext, params, function(user) {
+
+                // Verify the user doesn't need to verify their email address
+                RestAPI.User.getUser(camAdminRestContext, user.id, function(err, user) {
+                    assert.ok(!err);
+                    assert.strictEqual(user.email, email.toLowerCase());
+                    assert.strictEqual(user.emailVerified, true);
+                    assert.ok(!user.emailChange);
+
+                    // Assert that there's a mapping for the email address
+                    PrincipalsTestUtil.assertUserEmailMapping(email, [user.id], function() {
+                        return callback();
+                    });
+                });
+            });
+        });
+
+        /**
+         * Test that verifies that user accounts created through trusted external SSO sources do not need to verify their email address
+         */
+        it('verify user accounts created through trusted external SSO sources do not need to verify their email address', function(callback) {
```

Comment:
Despite all my rallying for more verifications, I agree that this does not need email verification from OAE :P